test/runtests.jl

```diff
@@ -15,44 +15,84 @@ using Aqua: Aqua
 """
     experimental_orders_of_convergence(prob, alg, dts;
                                       test_time = nothing,
-                                      only_first_index = false)
+                                      only_first_index = false,
+                                      ref_alg = TRBDF2(autodiff = false))
 
 Solve `prob` with `alg` and fixed time steps taken from `dts`, and compute
 the errors at `test_time`. If`test_time` is not specified the error is computed 
 at the final time.
 Return the associated experimental orders of convergence.
 
 If `only_first_index == true`, only the first solution component is used
-to compute the error.
+to compute the error. If no analytic solution is available, a reference
+solution is computed using `ref_alg`.
 """
 function experimental_orders_of_convergence(prob, alg, dts; test_time = nothing,
-                                            only_first_index = false)
+                                            only_first_index = false,
+                                            ref_alg = TRBDF2(autodiff = false))
     @assert length(dts) > 1
     errors = zeros(eltype(dts), length(dts))
```

Comment:
Could we maybe simplify the logic by avoiding duplicating a lot of code? Maybe start with something like

```suggestion
    errors = zeros(eltype(dts), length(dts))
    
    if !(isnothing(prob.f.analytic))
        # there is an analytic solution
        if isnothing(test_time)
            # we compare the results at the final time
            reference_solution = prob.f.analytic(prob.u0, prob.p, last(prob.tspan))
        else
            # we compare the results at the given time
            reference_solution = prob.f.analytic(prob.u0, prob.p, test_time)
        end
    else
        # we compute a reference solution numerically
        if isnothing(test_time)
            # we compare the results at the final time
            tspan = prob.tspan
        else 
            # we compare the results at the given time
            tspan = (first(prob.tspan), test_time)
        end
        dt0 = (tspan[end] - tspan[begin]) / 1e5
        refsol = solve(prob, ref_alg; 
                       dt = dt0, adaptive = false,
                       save_everystep = false)
        reference_solution = refsol.u[end]
   end
```

Then, we can use `reference_solution` in the checks below. This avoids duplicating some code and decouples the computation of the reference solution from the numerical solution we are interested in.